Flutter/datagrid/column-drag-and-drop.md

```diff
@@ -0,0 +1,338 @@
+
+{% tabs %}
+{% highlight Dart %}
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Syncfusion Flutter DataGrid')),
+      body: SfDataGrid(
+        source: employeeDataSource,
+        allowColumnsDragging: true,
+        columns: columns,
+        onColumnDragging: (DataGridColumnDragDetails details) {
+         if (details.action == DataGridColumnDragAction.update &&
+              details.to == 2) {
+            return false;
+          }
+          if (details.action == DataGridColumnDragAction.dropped &&
+              details.to != null) {
+            final GridColumn rearrangeColumn = columns[details.from];
+            columns.removeAt(details.from);
+            columns.insert(details.to!, rearrangeColumn);
+            employeeDataSource.buildDataGridRow();
+            employeeDataSource.updateDtaGrid();
+          }
+          return true;
+        },
+      ),
+    );
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+## Changing the feedback widget
+
+The DataGrid allows you to change the drag feedback widget by returning a custom widget from the `SfDataGrid.columnDragFeedbackBuilder` builder. This allows you to change the drag feedback widget according to your requirements.
+
+{% tabs %}
+{% highlight Dart %}
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(title: const Text('Syncfusion Flutter DataGrid')),
+      body: SfDataGrid(
+        source: employeeDataSource,
+        allowColumnsDragging: true,
+        columns: columns,
+        columnDragFeedbackBuilder: (context, column) {
+          return Container(
+            height: 50,
+            width: column.actualWidth,
+            color: Colors.grey,
+            child: const Center(
+              child: DefaultTextStyle(
+                style: TextStyle(
+                    fontSize: 14,
+                    color: Colors.pink,
+                    fontWeight: FontWeight.bold),
+                child: Text('Drag View'),
+              ),
+            ),
+          );
+        },
+        onColumnDragging: (DataGridColumnDragDetails details) {
+         if (details.action == DataGridColumnDragAction.dropped &&
+              details.to != null) {
+            final GridColumn rearrangeColumn = columns[details.from];
+            columns.removeAt(details.from);
+            columns.insert(details.to!, rearrangeColumn);
+            employeeDataSource.buildDataGridRow();
+            employeeDataSource.updateDtaGrid();
+          }
+          return true;
+        },
+      ),
+    );
+  }
+
+{% endhighlight %}
+{% endtabs %}
+
+<img alt="Flutter datagrid shows a checkbox filter in web platform" src="images/column-drag-and-drop/feedback-widget.gif" width="400"/>
+
+## Drag indicator customization
+
+The DataGrid allows you to customize the drag indicator by changing its color and thickness. The DataGrid should be wrapped inside the `SfDataGridTheme`.
```

Comment:
The color and thickness of the drag indicator can be customized by the SfDataGridThemeData.columnDragIndicatorColor and SfDataGridThemeData.columnDragIndicatorStrokeWidth properties.